src/cmd_line/commands/write.ts

```diff
@@ -50,7 +50,7 @@ export class WriteCommand extends node.CommandBase {
       return;
     }
 
-    if (vimState.editor.document.isUntitled) {
+    if (vimState.editor.document.isUntitled || vimState.editor.document.uri.scheme !== 'file') {
```

Comment:
Is this logic correct? If this is NOT a file, we want to save?

Is there an API we can instead leverage such that we can save for local and remote files?

Reply:
I will check whether there is such api.

Reply:
the issue lies in fs module which can only check the files with "file" schema.

Reply:
Would you mind adding a comment explaining this if clause? Maybe something along the lines of:

// defer saving the file to vscode if file is new (to present file explorer) or if file is a remote file

Reply:
What's the scheme on a remote file?

Reply:
I'm using the package "Remote FS". It provides two ways to access remote files “ftp” and "sftp".
sftp is based on ssh. 
If I uses "sftp" to access remote file, the file scheme would be "sftp".

there is another extension which called "memfs" which is used to create a totally memory based file system. This extension is provided by VSCode team as tutorial on VSCode's "FileSystemProvider".
The files created by this "memfs" has the scheme "memfs". VSCodeVim also has the issue on saving this file scheme.

The real issue for saving files with non-"file" scheme is that VSCodeVim tries to check the file existence by "fs" module of nodejs before calling the "save" API.
VSCode's save API can work on different file schemes, but fs module can't. It only support “file:” protocol. 
https://nodejs.org/api/fs.html#fs_url_object_support
![image](https://user-images.githubusercontent.com/2599606/50545760-348eb800-0c56-11e9-82f2-a91009f0e103.png)

I haven't found an common api from VSCode which can check file's existence to replace fs access API.
Is it OK to remove the access checking on non-"file" scheme files?

Reply:
From my review:

> LGTM. If you could add that documentation, that would be great.